onnxruntime/core/providers/cuda/tensor/scatter_elements_impl.cu

```diff
@@ -37,61 +64,162 @@ __global__ void _ScatterElementsKernel(
   output_data[data_idx] = updates[indices_index];
 }
 
+// From the innermost axis (largest) check equality of dim value of input and indices.
+// If same, merge it and continue. Otherwise, copy remaining. The scatter axis need
+// to be keep.
+static int CompactInputIndicesDims(
```

Comment:
For <=2D, this should have a very quick return.

Reply:
agree. As impact is low, will change it in future.